frontend/src/Pages/Schedule.tsx

```diff
@@ -63,85 +77,122 @@ export default function Schedule() {
         setSelectedEvent(null);
     }
 
+    function canUpdateEvent(): boolean {
+        if (!selectedEvent || selectedEvent.is_override) {
+            return false;
+        }
+        if (class_id && selectedEvent.class_id.toString() != class_id)
+            return false;
```

Comment:
@carddev81 Do you think we should make the buttons disabled if `canUpdateEvent` is going to return `false`? 

If we disable them we'd need to provide a tooltip, i.e., "This event is from another class and cannot be edited on the {class_name} schedule."

If {class_name} is going to be too long, then we could say, "This event is from another class and cannot be edited from here."